previewgif.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+# Required Environments Variables:
+#	DEMO_MAGIC_PATH: Path to demo-magic.sh
+GITGUD_REPO=$1
+
+# ===== Configure demo-magic.sh
+. $DEMO_MAGIC_INSTALL/demo-magic.sh
+PROMPT_TIMEOUT=2
+TYPE_SPEED=10
+DEMO_PROMPT="${GREEN}➜ ${CYAN}\W${RED}\$ "
+
+# ===== Configure Shell 
+OLD_HOME=$(pwd)
+ 
+# We want to exit if any command doesn't work, or if
+# an undefined variable is used.
+set -euo pipefail
+
+cd /tmp
+clear
+
+# ===== Begin presentation 
+p "pip3 install git-gud"
+pip install ${GITGUD_REPO}
+
+p "# We'll first make an empty directory for git gud."
+pe "mkdir my-git-gud"
+pe "cd my-git-gud"
+p "# Now, let's initialize!"
+pe "git gud init"
+wait || true
+
+p "# Here's a demonstration of some of the levels that we have in store for you!"
```

Comment:
This line ends up overflowing, is there any way to either change the dimensions? If not, maybe we should shorten this line

Reply:
@sahansk2 This is the main thing preventing a merge right now

Reply:
Ehhh, not sure. Pretty sure that you can't change it as it is, and that you'd have to use `tmux` or something like that, and record the `tmux` session, and, well....it's far easier to just split it into multiple lines.

`# Here are just some of the levels that you can try out!` is shorter.

Reply:
As long as it fits and makes sense, feel free to put whatever text you want here